lib/msprime.c

```diff
@@ -504,6 +520,9 @@ msp_alloc(msp_t *self,
     /* Set up pedigree */
     self->pedigree = NULL;
 out:
+    if (ret != 0 && self->recomb_map != NULL) {
```

Comment:
This shouldn't be necessary - any cleanup gets done in msp_free.

Reply:
Is it expected that `msp_free` should always be called should `msp_alloc` fail?

Reply:
That is currently not the case in the C unit tests.

Reply:
> Is it expected that msp_free should always be called should msp_alloc fail

Yes, that's the convention we ended up with in tskit --- the cleanup is too messy otherwise. Where do we do otherwise in the C unit tests? (Might be quicker to call down to talk this one through)